### PR TITLE
AGL Dashboard GH Repos

### LINKED ISSUE #31: User Dashboard: Github Repositories
**Details:** A user exists in the database. They have a Github token stored. Generate a token here for testing and development: https://github.com/settings/tokens

A future story will implement generating these tokens dynamically but that isn't necessary here.

```
As a logged in user
When I visit /dashboard
Then I should see a section for "Github"
And under that section I should see a list of 5 repositories with the name of each Repo linking to the repo on Github
```

- [x] API call uses: https://developer.github.com/v3/repos/#list-your-repositories
- [x] API call **DOES NOT** use: https://developer.github.com/v3/repos/#list-user-repositories
- [x] No hashes in the view.
- [x] Uses objects in the view to represent the Repos.
- [x] Edge case: Make sure this shows the proper repositories when there are more than one user in the database with different tokens.
- [x] Don't display the "Github" section if the user is missing a Github token

### PR DESCRIPTION
**Pull request by alex-latham**
Adds functionality to display five most recent GH repos on a user's dashboard if they have a GH token. Note: the new tests read two GITHUB_TOKENs from ENV. I added these to Travis CI so the tests should still work there.

New files: 
  - models/git_hub/repo.rb    <<< call model methods here like `GitHub::Repo`
  - services/git_hub_services.rb
 - spec/features/user/user_can_view_own_repos_spec.rb
    - Includes tests for all listed edge cases

Other:
 - Added a new ApplicationController method: #gh_token, which returns a user's GH token (or nil if there isn't one)
- Rubocop has no citations

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -21,4 +21,8 @@ def tutorial_name(id)
   def four_oh_four
     raise ActionController::RoutingError, 'Not Found'
   end
+
+  def gh_token
+    current_user.gh_token
```
Comment (TravisBorgsmiller) [inline app/controllers/application_controller.rb:26]: I think this will be changed in the next PR

File: spec/features/user/user_can_view_own_repos_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    user = create(:user)
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    visit dashboard_path
+
+    expect(page).not_to have_selector('#github')
+  end
+
+  it 'cannot see other registered users GitHub repos' do
+    user1 = create(:user)
```
Comment (TravisBorgsmiller) [inline spec/features/user/user_can_view_own_repos_spec.rb:32]: Solid looking sad path testing

**Review by TravisBorgsmiller — APPROVED** [review]
